### ISSUE TITLE
Segfault in subset of quasar models

### ISSUE THREAD

**Issue by jhmatthews** [sirocco-rt/sirocco#973]
I'm running a grid of quasar models on the Cambridge CSD3 cluster ([cascade lake nodes](https://docs.hpc.cam.ac.uk/hpc/user-guide/cclake.html)). 14 out of 54 of the models fail with segmentation faults. The backtrace is

```
==== backtrace (tid: 129222) ====
 0 0x000000000041ca5e kappa_ff()  ???:0
 1 0x000000000041bdf5 calculate_ds()  ???:0
 2 0x000000000040d252 translate_in_wind()  ???:0
 3 0x000000000040d6e8 translate()  ???:0
 4 0x0000000000419859 trans_phot_single()  ???:0
 5 0x000000000041a17e trans_phot()  ???:0
 6 0x000000000046d96e calculate_ionization()  ???:0
 7 0x00000000004050b9 main()  ???:0
 8 0x0000000000022555 __libc_start_main()  ???:0
 9 0x0000000000405517 _start()  ???:0
=================================
``` 

It's possible this could be related to #885 which seems to report a similar backtrace. 

**Versions:** I was running with an old version of python so that I was using the same version Jago used for his simulations. I used commit f840486e78166d55a65e9ef345ef8035dfc471d8 for the initial grid, but I've checked the same segfault occurs with the latest dev version dba94730901f0724b965994fdab707c3da6df6f1. My MPI/gcc environment is:
```
$ mpicc --version
gcc (Spack GCC) 9.3.0

$ mpirun --version
mpirun (Open MPI) 4.0.4
```
I will try these files on my mac and see if they also fail then.

**Comment by jhmatthews** [comment]
Ok, so running the same model on my mac does not produce a segfault. There does however look to be something odd going in in cell 934. This cell produces various errors:
```
wind2d: Cell 934 ( 9,34) in domain 0 has 1 corners in wind, but zero volume
calculate_ds: frequency along photon 123160 path's in cell 934 (nplasma 885) is the same (dfreq=0.00e+00)
ds_calculate: wind vol = 0 for cell 934 (nplasma 885 inwind -2) photon position -8.84187e+16 -2.00325e+16 -4.63708e+16
```

It seems to be flagged as W_IGNORE which happens as a result of the first error. It also has zero density. However, it still seems to have photons translated through it and is trying to increment estimators and so on. The part of the code that deals with whether the photons are in the wind is really pretty hard to follow so I'm struggling to figure this out at the moment.

**Comment by jhmatthews** [comment]
I'm not entirely sure I've got the cause of the segfault yet, but I have confirmed on my mac that photons are entering a cell that is flagged as W_IGNORE and trying to increment estimators and so on. My guess is this is the cause of the segfault but it's compiler/architecture sensitive. I've raised a separate issue to deal with code logic in #974.

**Comment by jhmatthews** [comment]
A change of how smax is dealt with for W_IGNORE cells seems to stop the segfault. Specifically, as an attempted fix I removed the move_phot statement from smax_in_cell(), and changed the code logic slightly so the estimator routines are not entered if the cell is flagged as W_IGNORE. None of this is in dev yet as it would be good to resolve some of the questions outlined in #974 first

**Comment by kslong** [comment]
With the changes that James made, this issue should be solved.
